controllers/users-controllers.js

```diff
@@ -7,3 +10,19 @@ exports.getAllUsers = (req, res, next) => {
     })
     .catch(next);
 };
+
+exports.getUsernameByUsername = (req, res, next) => {
+  const { username } = req.params;
+  if (typeof username !== "string"|| !isNaN(username)) {
+    return res.status(400).send({ msg: "incorrect input" });
+  }
+  selectUsernameByUsername(username)
```

Comment:
It's not the responsibility of a controller function to validate data, it's a model's responsibility, I don't imagine that this data could be invalid if it's just any combination of string characters